soh/src/code/z_draw.c

```diff
@@ -827,12 +827,12 @@ void GetItem_DrawGenericMusicNote(GlobalContext* globalCtx, s16 drawId) {
 
     OPEN_DISPS(globalCtx->state.gfxCtx);
 
-    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx,  __FILE__, __LINE__), G_MTX_MODELVIEW | G_MTX_LOAD);
-    gsDPSetGrayscaleColor(POLY_OPA_DISP++, colors[color_slot][0], colors[color_slot][1], colors[color_slot][2], 255);
-    gsSPGrayscale(POLY_OPA_DISP++, true);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx,  __FILE__, __LINE__), G_MTX_MODELVIEW | G_MTX_LOAD);
+    gsDPSetGrayscaleColor(POLY_XLU_DISP++, colors[color_slot][0], colors[color_slot][1], colors[color_slot][2], 255);
+    gsSPGrayscale(POLY_XLU_DISP++, true);
     func_80093D18(globalCtx->state.gfxCtx);
-    gSPDisplayList(POLY_OPA_DISP++, sDrawItemTable[drawId].dlists[0]);
-    gsSPGrayscale(POLY_OPA_DISP++, false);
+    gSPDisplayList(POLY_XLU_DISP++, sDrawItemTable[drawId].dlists[0]);
+    gsSPGrayscale(POLY_XLU_DISP++, false);
```

Comment:
Resolves #1434

Reply:
does this touch/fix anything outside of shopsanity? if so it might be worth breaking out into a separate PR

Reply:
#1434 mostly affects shopsanity, the only other known instance of it happening was in the back of GTG, the dark door overlay was rendered in front of the song item. In shopsanity pretty much all shelves were rendered in front of these songs making them effectively invisible. 

I still see the argument for splitting it out though, if we want I can do that

Reply:
let's split it out then, even just going into rando-next as a quick PR will be nice (it'll be easier to track down the change in the future that way)

Reply:
I know you already am splitting this out now, but this touches everywhere song notes are drawn. So this goes for freestanding items like PoH's and freestanding keys (like in GtG). It's also used when you get an item and it's held above your head. If it turns out to be a problem in those cases, it might be good to look at a way to use a custom draw function just for songs in shops.

Reply:
> I know you're already splitting this out now, but this touches everywhere song notes are drawn. So this goes for freestanding items like PoH's and freestanding keys (like in GtG). It's also used when you get an item and it's held above your head. If it turns out to be a problem in those cases, it might be good to look at a way to use a custom draw function just for songs in shops.

great call, once the split off PR is up let's test those cases on that branch

Reply:
We had an ongoing conversation about this in #bug-reports, but basically it was discovered that this change makes non-warp songs behave more like warp songs and other freestanding items like PoH. This is not perfect because warp songs and PoH have their own graphical issues even in vanilla[1]. So while not perfect it's better in basically every way to have the non-warp songs behave like warp songs, rather than them behave different and have different sets of issues.

[1]
<img width="269" alt="Screen_Shot_2022-09-07_at_11 49 05_AM" src="https://user-images.githubusercontent.com/7316699/190443525-1ff6f950-03af-4ea1-a59d-517c62c22bbd.png">

Reply:
Yeah those types of situations were already pretty well tested back when this change was initially made, because I had the same thought. Seems like the songs always seem to either be in front of or behind stuff they shouldn't be, but as we can see that applies to vanilla items using the POLY_XLU functions too. Might even be like that on N64, but even if it wasn't that means the problem is deeper than just the notes, so this fix should be fine.

Reply:
#1488